### DISCUSSION TITLE
Hydra and Mysql with master/master replication

### DISCUSSION THREAD

**Discussion by oloake** [ory/hydra#2765] [Q&A] [answered]
Hi,

I would like to know if it is safe to use Hydra together with MySql in a master/master setup where writes can happen on either master and are replicated to the other. 

I found one answer in this thread which suggests that it should work:

_Of course you can do that.
We use uuid, so collisions are incredibly unlikely:_
https://community.ory.sh/t/high-availability-and-scaling/664/12

However that was a couple of years ago and I see that some tables use auto-incremented primary keys (in 1.10.2 at least) which require special care in a replicated setup. 

So does anyone have experience running Hydra with Mysql master/master and can tell me if this is a viable option?

Thanks!
Olof

**Comment by aeneasr** [comment, marked as answer]
Hello! Nice find :) @grantzvolsky is actually refactoring this at this moment. We plan on releasing a better database model that is also easier to clean up and scale step by step over the next months.

**Reply by oloake** [reply]
That sounds great. Thanks!
